Baikal/ML/inference_impl.cpp

```diff
@@ -0,0 +1,42 @@
+#include "inference_impl.h"
+
+#include <numeric>
+#include <valarray>
+
+
+namespace Baikal::ML
+{
+    namespace
+    {
+        Buffer AllocBuffer(const Buffer::Shape& shape)
+        {
+            size_t size = std::accumulate(shape.begin(),
```

Comment:
and what about multi-channel image ?

Reply:
It is actually expected that the input shape contains the number of channels as the last dimension. So for the output shape it is overridden with 3.
May be it's not the best idea though.